app/src/androidTest/java/com/example/evently/ViewNotificationsTest.java

```diff
@@ -0,0 +1,251 @@
+        // Any notifications sent to the All channel for participated event IDs should show up.
+        // See the setUpNotifications to figure out which notifications we're expecting here.
+        final var expectedNotifications = new Notification[] {
+            templateNotification(1, Notification.Channel.All),
+            templateNotification(3, Notification.Channel.All),
+            templateNotification(7, Notification.Channel.All)
+        };
+
+        // For each of the expected notifications, scroll to it and make sure it shows properly.
+        //        for (final var expectedNotification : expectedNotifications) {
+        //            assertRecyclerViewItem(
+        //                    R.id.notif_list,
+        //                    p(R.id.notif_title, expectedNotification.title()),
+        //                    p(R.id.notif_description, expectedNotification.description()));
+        //        }
+        onView(withId(R.id.notif_list)).check(matches(isDisplayed()));
+    }
+
+    @Test
+    public void expectNotification_winner() throws InterruptedException {
+        // Notifications sent to the Winner channel for won event IDs should show up.
+
+    }
+
+    @Test
+    public void expectNotification_loser() throws InterruptedException {
+        // Any notifications sent to the Loser channel for lost event IDs should show up.
+
+    }
+
+    @Test
+    public void acceptInvitation_markSeen() throws InterruptedException {
+        // Accepting an invitation should put self in the accepted list of entrants
+        // and mark the notification as seen.
+
+    }
+
+    @Test
+    public void declineInvitation_markSeen_expectNotification_cancelled()
+            throws InterruptedException {
+        // Declining an invitation should put self in the cancelled list of entrants
+        // and mark the notification as seen. Also expose any cancelled channel notifications.
+
+    }
+
+    @Test
+    public void noExpectNotification_otherEvent() throws InterruptedException {
+        // Notifications from non-participated event should NOT show up.
+    }
+
+    @Test
+    public void noExpectNotification_otherChannel() throws InterruptedException {
+        // Notifications from participated event but non-participated channel should NOT show up.
+
+    }
+
+    // Helper function to create a notification from the event index and target channel.
+    private static Notification templateNotification(Integer idx, Notification.Channel channel) {
+        return new Notification(
+                mockEvents[idx].eventID(),
+                channel,
+                channel + " channel " + idx,
+                "Description " + idx);
+    }
+
+    @Override
+    protected int getGraph() {
+        return R.navigation.entrant_graph;
+    }
+
+    @Override
+    protected Class<ViewNotificationsFragment> getFragmentClass() {
+        return ViewNotificationsFragment.class;
+    }
```

Comment:
You must implement these two methods as per the fragment you're testing.

It's pretty straight forward. `getFragmentClass` should ideally be auto gen'd because it's literally just `<T>.class` but Java is far too weak of a language to allow that.

The graph is entrant_graph for any entrant fragments. organizer_graph for any organizer fragments.